### ISSUE TITLE
TypeORM - Cannot determine type of empty array

### ISSUE THREAD

**Issue by linsolas** [oguimbal/pg-mem#131]
Hello,

I have an application where I use Postgresql with TypeORM. I want to test this application with a memory DB, so I use **pg-mem**.

In my Entity, I have a column, `tags`, which is defined as an array of strings:

```ts
import { Column, CreateDateColumn, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity()
export class Playground {

  @PrimaryGeneratedColumn() 
  id: number;

  @Column()
  shortId: string;

  @Column()
  name: string;

  // HERE
  @Column('varchar', { array: true })
  tags: string[];

  @Column({ default: false })
  archived: boolean;

  @CreateDateColumn()
  creationDate: Date;
}
```

With PG / TypeORM, I don't have any issue to insert an item, but in my unit tests, pg-mem says it cannot determine the type of the empty array:

> QueryFailedError: cannot determine type of empty array
>
> INSERT INTO "playground"("shortId", "name", "tags", "archived", "creationDate") VALUES ('TSAX-aslJ', 'Test', ARRAY[], DEFAULT, '2021-06-21T09:15:11.928Z') RETURNING "id", "creationDate";

If I modify the code to replace an empty array by `'{}'`, it is working:

> INSERT INTO "playground"("shortId", "name", "tags", "archived", "creationDate") VALUES ('TSAX-aslJ', 'Test', '{}', DEFAULT, '2021-06-21T09:15:11.928Z') RETURNING "id", "creationDate";

Am I missing something here?

### Technical info

Versions used:

* `"pg": "^8.6.0"`
* `"typeorm": "0.2.29"` - I used the last one (`0.2.32`) but downgraded to `0.2.29` due to https://github.com/oguimbal/pg-mem/issues/81
* `"pg-mem": "^1.9.9"`
* `"jest": "24.9.0"`
* node version `12.14.1`

pg-mem created this way on my tests:

```ts
const db = newDb({
  autoCreateForeignKeyIndices: true
});
const connection = await db.adapters.createTypeormConnection({
  type: 'postgres',
  entities: [Playground],
  synchronize: true
});
```

Let me know if you need more information.

Thanks.

**Comment by oguimbal** [comment]
Could you try again ? (it may have been fixed by #136 )

**Comment by linsolas** [comment]
Hello @oguimbal 

I've checked the version `1.9.10` and the problem is now fixed.

Thanks a lot.
